app/src/main/java/ru/yandex/practicum/contacts/presentation/sort/SortTypeUI.java

```diff
@@ -22,6 +23,11 @@ public boolean isSelected() {
         return selected;
     }
 
+    @Override
+    public boolean theSameAs(SortTypeUI other) {
+        return this.hashCode() == other.hashCode();
```

Comment:
Здесь для корректной работы сортировки необходимо сравнивать поля `sortType`:
```suggestion
        return this.getSortType() == other.getSortType();
```